docs/smif-prerequisites.md

```diff
@@ -1,27 +1,341 @@
+| --- | --- | --- | --- |
+| model name | string | `energy_demand` | A unique name
+| path | string | `../../models/energy_demand/run.py` | Relative to the project folder |
+| classname | string | `EnergyDemandWrapper` | Name of the python class in the wrapper file |
+| inputs | list | \<see below\> | |
+| outputs | list | \<see below\> | |
+| parameters | list | \<see below\> | |
+| interventions | string | `energy_demand.yml` | Name of the interventions file in the `project/data/interventions` folder |
+| initial conditions | string | `energy_demand_existing.yml` | Name of the file |
+
+### Inputs and Outputs
+
+`smif` requires all model inputs and outputs to be explicitly defined
+so that data can be passed to and retrieved from a model at runtime.
+
+Wherever you have a data passed into your model from another source, you should
+define an input.
+For each of the results you model produces, you need to define an output.
+For example, a digital communications model may require `population` data as an
+input, and produce a `service quality` metric and 
+an `fibre-optic repeater electricity demand` as an output
+
+| Attribute | Type | Example | Notes |
+| --- | --- | --- | --- |
+| name | string | `population_density` | Unique to inputs or outputs |
+| spatial_resolution | string | `lad` | Reference to the name of a region definition |
+| temporal_resolution | string | `annual` | Reference to the name of an interval definition |
+| units | string | `people/km^2`| SI units are automatically parsed, otherwise a warning is raised |
+
+The dependency upon another data source are explicitly declared in the
+integration framework.
+To declare a dependency, both models must have the requisite inputs
+and outputs defined.
+For example, if you wish to couple your energy demand model
+with an energy supply model, you will need to define outputs
+for `electricity demand`, `natural gas demand`, `hydrogen demand` and so on.
+The energy supply model would then need to define inputs for `electricity`,
+`natural gas` and `hydrogen`.
+Note that the names should be unique within a sector model and list of
+inputs and outputs.
+It is helpful if the names are easy to understand or descriptive.
+
+### Parameters
+
+Parameters are the means by which the 'dials and knobs' of a model can be made
+visible to `smif`. Once defined, parameters can be modified using narratives,
+and included in the smif GUI.
+
+Initially, (as of `smif v0.5`) only floating point parameters are supported,
+but future versions will support categorical and boolean parameters.
+
+| Attribute | Type | Example | Notes |
+| --- | --- | --- | --- |
+| name | string | `assump_diff_floorarea_pp`| |
+| description | string | `Difference in floor area per person in end year compared to base year` | |
+| absolute_range | tuple | (0.5, 2) | |
+| suggested_range | tuple | (0.5, 2) | |
+| default_value | float | 1 | |
+| units | string | `percentage` | |
+
+## Data Requirements
+
+### Initial Conditions
```

Comment:
Do we need to handle two things here?
- Initial system: where the modelled system can be built from a list of interventions (e.g. mobile digital communications model), this should be equivalent to a pipeline of interventions, where build dates are all from timesteps before the base year. Where this is not so clear or technically simple (e.g. water supply), this could be empty.
- Initial conditions: initialisation of parameters which are otherwise dynamically determined by the model (elsewhere called 'state').